Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -332,116 +277,62 @@ open class NavigationMapView: UIView {
-        }
-    }
-    
     /**
      Updates the map view’s preferred frames per second to the appropriate value for the current route progress.
      
-     This method accounts for the proximity to a maneuver and the current power source. It has no effect if `tracksUserCourse` is set to `true`.
+     This method accounts for the proximity to a maneuver and the current power source.
+     It has no effect if `NavigationCameraState` is in `.following` mode.
      */
-    open func updatePreferredFrameRate(for routeProgress: RouteProgress) {
-        guard tracksUserCourse else { return }
+    public func updatePreferredFrameRate(for routeProgress: RouteProgress) {
+        guard navigationCamera.state == .following else { return }
         
         let stepProgress = routeProgress.currentLegProgress.currentStepProgress
         let expectedTravelTime = stepProgress.step.expectedTravelTime
         let durationUntilNextManeuver = stepProgress.durationRemaining
         let durationSincePreviousManeuver = expectedTravelTime - durationUntilNextManeuver
-        let conservativeFramesPerSecond = UIDevice.current.isPluggedIn ? FrameIntervalOptions.pluggedInFramesPerSecond : minimumFramesPerSecond
-        
-        if let upcomingStep = routeProgress.currentLegProgress.upcomingStep,
-           upcomingStep.maneuverDirection == .straightAhead || upcomingStep.maneuverDirection == .slightLeft || upcomingStep.maneuverDirection == .slightRight {
-            mapView.preferredFPS = shouldPositionCourseViewFrameByFrame ? FrameIntervalOptions.defaultFramesPerSecond : conservativeFramesPerSecond
-        } else if durationUntilNextManeuver > FrameIntervalOptions.durationUntilNextManeuver &&
-                    durationSincePreviousManeuver > FrameIntervalOptions.durationSincePreviousManeuver {
-            mapView.preferredFPS = shouldPositionCourseViewFrameByFrame ? FrameIntervalOptions.defaultFramesPerSecond : conservativeFramesPerSecond
+        
+        let maneuverDirections: [ManeuverDirection] = [.straightAhead, .slightLeft, .slightRight]
+        if let maneuverDirection = routeProgress.currentLegProgress.upcomingStep?.maneuverDirection,
+           maneuverDirections.contains(maneuverDirection) ||
+            (durationUntilNextManeuver > FrameIntervalOptions.durationUntilNextManeuver &&
+                durationSincePreviousManeuver > FrameIntervalOptions.durationSincePreviousManeuver) {
+            mapView.preferredFPS = UIDevice.current.isPluggedIn ? FrameIntervalOptions.pluggedInFramesPerSecond : minimumFramesPerSecond
         } else {
-            mapView.preferredFPS = FrameIntervalOptions.pluggedInFramesPerSecond
+            mapView.preferredFPS = FrameIntervalOptions.defaultFramesPerSecond
         }
     }
     
-    /**
-     Track position on a frame by frame basis. Used for first location update and when resuming tracking mode
-     */
-    public func enableFrameByFrameCourseViewTracking(for duration: TimeInterval) {
```

Comment:
From now on we no longer provide temporary frame by frame updates to `UserCourseView`, it's always updated frame by frame.